### ISSUE TITLE
[Accessibility]: Highcharts Graphs are not adapting Windows High contrast Theme.

### ISSUE THREAD

**Issue by Shubham786786** [highcharts/highcharts#15567]
#### User Impact
User who use High Contrast theme will be impacted here as they will not grab information given in the Graphs as graphs are not adapting high Contrast Theme. 

#### Expected behaviour
Graphs should adapt high contrast Theme.

#### Actual behaviour
Graphs are not adapting high contrast Theme.

#### Test Environment:
OS: Windows 10 Version Dev (OS Build 21343.1000)
Browser: Microsoft Edge Dev Build 91.0.864.1
URL: [https://www.highcharts.com/demo/pie-legend](url)

#### Steps to reproduce
1. Open https://www.highcharts.com/demo/pie-legend on browser.
2. Open Windows Settings app using 'Windows + i' Key.
3. Navigate to 'Ease of Access' menu item and activate it.
4. Then Navigate to 'High Contrast' tab and activate it.
5. Then switch ON 'Turn on high Contrast' toggle button.
6. Select White/Black theme from 'Choose a theme' combo box. 
7. Verify that Graphs are adapting high contrast themes or not.

#### MAS Reference
[MAS.4.3.1.No.Disruption.of.Accessibility.Features.docx](https://github.com/highcharts/highcharts/files/6363945/MAS.4.3.1.No.Disruption.of.Accessibility.Features.docx)

#### Attachments
[Attachments Files.zip](https://github.com/highcharts/highcharts/files/6363969/Attachments.Files.zip)

**Comment by KacperMadej** [comment]
Hi @Shubham786786 

Thank you for reporting the problem.
Windows High Contrast Theme's dynamic change is currently not supported.

_Internal note:_
@oysteinmoseng Looks like this could be added in [a rather simple way](https://developer.mozilla.org/en-US/docs/Web/API/Window/devicePixelRatio#monitoring_screen_resolution_or_zoom_level_changes).

**Comment by oysteinmoseng** [comment]
_Internal note:_ Agree @KacperMadej - we should add a listener to the change event on the media query, like in that example.

**Comment by VivianSi233** [comment]
Bug is repro on bar chart(please see image from link below), 
pie chart(https://github.com/highcharts/highcharts/issues/15040), 
area chart(https://github.com/highcharts/highcharts/issues/15921)
![image](https://user-images.githubusercontent.com/89379019/138205746-bf3fe1ba-44ad-4bde-8479-a06a3397f623.png)

**Comment by oysteinmoseng** [comment]
@VivianSi233 Hi, your image is currently intended behavior, but we do want to allow setting specific colors for series in High contrast mode, as referenced in #15921.

Note however [WCAG 1.4.1](https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-without-color.html), use of color should not be the only means of conveying information such as series name in a bar chart.

**Comment by VivianSi233** [comment]
@oysteinmoseng Do you have some workaround for this bug? I received similar bugs which caused by this bug, so leave this message to highlight.

**Comment by oysteinmoseng** [comment]
Hi @VivianSi233, here is a simple workaround that makes Highcharts react when High contrast mode is activated: https://jsfiddle.net/oysteinmoseng/Lu3h6qzs/.

If you want it to also react when High Contrast Mode is deactivated you currently have to recreate the chart with something like this: https://jsfiddle.net/oysteinmoseng/tdq367vr/. This can be tricky because Highcharts can sometimes modify the input data, so you would have to keep that in mind.

**Comment by huangzumeihan** [comment]
According to your solution, the column with drilldown does not seem to work
![image](https://user-images.githubusercontent.com/31198399/172747811-4fad348a-89d0-4251-8716-2e1696b8ba5e.png)

**Comment by raf18seb** [comment]
Hi @huangzumeihan,

Stacked bar is working fine the same like the above pie demo when turning on the contrast mode: https://jsfiddle.net/BlackLabel/52jvpuo9/
However, @oysteinmoseng, it's not working for me when disabling it.

**Comment by VivianSi233** [comment]
When I open https://jsfiddle.net/BlackLabel/52jvpuo9/ and then apply contrast theme Aquatic, please see screenshot about actual result. We got another bug which reported on Win11 OS about 4 new contrast themes, mentioned that line chart does not adapt to High Contrast windows themes. 
![image](https://user-images.githubusercontent.com/89379019/190537828-b2590ad9-06ff-475a-aeca-81a11a7af218.png)

**Comment by VivianSi233** [comment]
This bug is repro on https://www.highcharts.com/demo/area-negative.
Low vision users will be impacted as graph line color are not visible in properly high contrast Aquatic/Desert mode.
Could you please help to investigate? thanks.
![image](https://user-images.githubusercontent.com/89379019/205543618-9a8143cd-fcc9-477e-a241-e409c0d40653.png)

**Comment by oysteinmoseng** [comment]
@VivianSi233 The lines in the screenshot above look correct to me. Do I understand correctly that you would prefer to override the high contrast white with your own colors to have different colors for each line?

**Comment by VivianSi233** [comment]
> @VivianSi233 The lines in the screenshot above look correct to me. Do I understand correctly that you would prefer to override the high contrast white with your own colors to have different colors for each line?

Low vision users will be impacted as graph line color for 'Total production' 'Gross consumption' and 'Trade surplus' are not visible in properly high contrast Aquatic/Desert mode. Since chart legends cannot give different guide for each part.

**Comment by VivianSi233** [comment]
@oysteinmoseng Based on your comments (please see screenshot below), I find legends from https://www.highcharts.com/demo/highcharts/line-chart have different shapes. So is it possible that legends from column chart have the same design which support different shapes? Such as https://www.highcharts.com/demo/highcharts/column-basic.
![image](https://github.com/user-attachments/assets/fa5d66cb-c8f2-48c8-99ef-e687f03ce760)

**Comment by VivianSi233** [comment]
@oysteinmoseng We find the **tooltip** can help users to get info from chart, it is not color alone to identify the column bars, right? Maybe don't need to add shapes for legends if tooltip can help users to identify the column bars. 
![image](https://github.com/user-attachments/assets/21048298-9a93-472f-9a14-8ba71a6ea7f2)

**Comment by VivianSi233** [comment]
[bugfix/15921-fillcolor-hcm by goransle · Pull Request #20703 · highcharts/highcharts (github.com)](https://github.com/highcharts/highcharts/pull/20703) **only works for area chart** but not column bar chart.

**Comment by oysteinmoseng** [comment]
Hi @VivianSi233, if you wanted to differentiate column series in the legend without color, your best bet would be using [pattern fills](https://www.highcharts.com/docs/chart-design-and-style/pattern-fills). It can be argued that tooltips offer another way to differentiate the columns, although since it is hidden behind interaction it is not optimal. It can also in some cases be argued that the visual order of the legend items correspond to the visual order of columns in the chart, as in your example above, and that this is another way to link columns with their names without color.
